lib/src/cf/mod.rs

```diff
@@ -0,0 +1,7 @@
+pub(crate) mod gc;
+pub(crate) mod reader;
+pub(crate) mod writer;
+
+pub use self::gc::*;
+pub use self::reader::read;
+pub use self::writer::Writer;
```

Comment:
This is what we use for recording changes.

More concretely, our `Transaction` now holds a reference to a single instance of this `Writer` and each create/update/delete to a document results in a function call to the Writer instance for "buffering changes" until the commit.